File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.java
class Solution {
    public int findPairs(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int num: nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        int count = 0;
        for (Map.Entry<Integer, Integer> pair: map.entrySet()) {
            if (k == 0) {
                if (pair.getValue() >= 2) {
                    count++;
                }
            } else {
                if (map.containsKey(pair.getKey() + k)) {
                    count++;
                }
            }
        }
        
        return count;
    }
}

File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        counter = Counter(nums)
        count = 0
        
        for num in counter:
            if k == 0 and counter[num] > 1 or k != 0 and k + num in counter:
                count += 1
        
        return count